python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py

```diff
@@ -0,0 +1,388 @@
+            leading_dimension,
+            dtype=th.float32,
+            device=ctx.device,
+        )
+        if coeff is None:
+            agg_fwd_func(agg_output, feat.detach(), mfg)
+        else:
+            agg_fwd_func(
+                agg_output,
+                feat.detach(),
+                mfg,
+                weights_combination=coeff.detach()
+            )
+
+        ctx.save_for_backward(feat, coeff)
+        return agg_output
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        r"""Compute the backward pass of R-GCN aggregation layer.
+
+        Parameters
+        ----------
+        grad_output : torch.Tensor, dtype=torch.float32
+            Gradient of loss function w.r.t output.
+        """
+        feat, coeff = ctx.saved_tensors
+
+        if ctx.idtype == th.int32:
+            agg_bwd_func = agg_hg_basis_post_bwd_int32
+        else:
+            agg_bwd_func = agg_hg_basis_post_bwd_int64
+
+        grad_feat = th.empty_like(feat, dtype=th.float32, device=ctx.device)
+        if coeff is None:
+            grad_coeff = None
+            agg_bwd_func(grad_feat, grad_output, feat.detach(), ctx.mfg)
+        else:
+            grad_coeff = th.empty_like(
+                coeff, dtype=th.float32, device=ctx.device
+            )
+            agg_bwd_func(
+                grad_feat,
+                grad_output,
+                feat.detach(),
+                ctx.mfg,
+                output_weight_gradient=grad_coeff,
+                weights_combination=coeff.detach()
+            )
+
+        return None, None, None, None, grad_feat, grad_coeff
+
+
+class CuGraphRelGraphConv(nn.Module):
+    r"""An accelerated relational graph convolution layer using the
+    highly-optimized aggregation primitives in cugraph-ops.
+    See :class:`dgl.nn.pytorch.conv.RelGraphConv` for mathematical model.
+
+    .. note::
+        This is an **experimental** feature.
+        Compared with :class:`dgl.nn.pytorch.conv.RelGraphConv`, this model:
+
+        * Only works on cuda devices.
+        * Only support basis-decomposition regularization.
+        * Requires an extra argument `fanout` as input, since the current model is
+          designed for sampled-graph (message-flow-graph) use-cases.
+          Full-graph support will be added in upcoming releases.
+
+    Parameters
+    ----------
+    in_feat : int
+        Input feature size.
+    out_feat : int
+        Output feature size.
+    num_rels : int
+        Number of relations.
+    fanout : int
+        Maximum number of sampled neighbors of an destination node;
+        i.e, maximum in degree of destination nodes
+    regularizer : str, optional
+        Which weight regularizer to use ("basis" or ``None``):
+         - "basis" is for basis-decomposition.
+         - ``None`` applies no regularization.
+        Default: ``None``.
+    num_bases : int, optional
+        Number of bases. Only used when "basis" regularizer is applied.
```

Comment:
```suggestion
        Number of bases. It comes into effect when the "basis" regularizer is applied. Otherwise, it uses number of relations (``num_rels``).
```

Reply:
> Otherwise, it uses number of relations (``num_rels``).

This is not the case in the current setup. The value is not being used when regularizer is None. I did add "It comes into effect when the "basis" regularizer is applied." for consistency with RelGraphConv.

Reply:
I see. Thanks.